src/pkg/util/nvidiautils/nvidiautils.go

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package nvidiautils
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/sylabs/singularity/src/pkg/sylog"
+)
+
+// generate bind list using nvidia-container-cli
+func nvidiaContainerCli() ([]string, []string, error) {
+	var strArray []string
+	var bindArray []string
+
+	// use nvidia-container-cli (if present)
+	command, err := exec.LookPath("nvidia-container-cli")
+	if err != nil {
+		return nil, nil, fmt.Errorf("no nvidia-container-cli present: %v\n", err)
+	}
+
+	// process the binaries first
+	cmd := exec.Command(command, "list", "--binaries")
+	out, err := cmd.Output()
+	if err != nil {
+		return nil, nil, fmt.Errorf("Unable to execute nvidia-container-cli: %v\n", err)
+	}
+
+	for _, line := range strings.Split(string(out), "\n") {
+		if line != "" {
+			// add this in to the bind list here so we don't need to special case it
+			// when the libraries are processed later in GetNvidiaBindPath
+			// (i.e. thiese will never show up in ldconfig output, hence they would not be
+			// added in later without adding a lot of special case code)
+			bindString := line + ":" + line
+			bindArray = append(bindArray, bindString)
+		}
+	}
+
+	cmd = exec.Command(command, "list", "--ipcs", "--libraries")
+	out, err = cmd.Output()
+	if err != nil {
+		return nil, nil, fmt.Errorf("Unable to execute nvidia-container-cli: %v\n", err)
+	}
+
+	for _, line := range strings.Split(string(out), "\n") {
+		if line != "" {
+
+			fileName := filepath.Base(line)
```

Comment:
I'm not sure this will work when it gets input that does not include `/` or does include whitespace or whatever, but I cannot test b/c I don't have access to a GPU right now.